cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunnerTest.java

```diff
@@ -0,0 +1,187 @@
+ */
+@PrepareForTest(TaskUtil.class)
+@RunWith(PowerMockRunner.class)
+public class RemoteSourceControlOperationRunnerTest {
+
+  private static CConfiguration cConf;
+  private static NettyHttpService httpService;
+  private static RemoteClientFactory remoteClientFactory;
+  private static MetricsCollectionService metricsCollectionService;
+  private static PushAppContext mockPushContext;
+  private static InMemorySourceControlOperationRunner mockInMemoryOperationRunner;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    // Mock the PushContext
+    RepositoryConfig mockRepoConfig = new RepositoryConfig.Builder()
+      .setProvider(Provider.GITHUB)
+      .setLink("ignored")
+      .setDefaultBranch("develop")
+      .setPathPrefix("pathPrefix")
+      .setAuthType(AuthType.PAT)
+      .setTokenName("gittoken")
+      .build();
+    ApplicationDetail mockAppDetails = new ApplicationDetail(
+      "app1", "v1", "description1", null, null, "conf1", new ArrayList<>(),
+      new ArrayList<>(), new ArrayList<>(), null, null);
+    CommitMeta mockCommit = new CommitMeta("author", "commiter", System.currentTimeMillis(), "commit");
+    mockPushContext = new PushAppContext(NamespaceId.DEFAULT, mockRepoConfig, mockAppDetails, mockCommit);
+
+    // Set up CConf and necessary services
+    cConf = CConfiguration.create();
+    cConf.setInt(Constants.TaskWorker.CONTAINER_KILL_AFTER_REQUEST_COUNT, 0);
+
+    metricsCollectionService = new NoOpMetricsCollectionService();
+    InMemoryDiscoveryService discoveryService = new InMemoryDiscoveryService();
+    remoteClientFactory = new RemoteClientFactory(discoveryService,
+                                                  new DefaultInternalAuthenticator(new AuthenticationTestContext()));
+    httpService = new CommonNettyHttpServiceBuilder(cConf, "test", new NoOpMetricsCollectionService())
+      .setHttpHandlers(
+        new TaskWorkerHttpHandlerInternal(cConf, discoveryService, discoveryService, className -> {
+        }, new NoOpMetricsCollectionService())
+      )
+      .setChannelPipelineModifier(new ChannelPipelineModifier() {
+        @Override
+        public void modify(ChannelPipeline pipeline) {
+          pipeline.addAfter("compressor", "decompressor", new HttpContentDecompressor());
+        }
+      })
+      .build();
+
+    // Start the http service that runs TaskWorkerHttpHandler and register services
+    httpService.start();
+    discoveryService.register(URIScheme.createDiscoverable(Constants.Service.TASK_WORKER, httpService));
+    discoveryService.register(URIScheme.createDiscoverable(Constants.Service.SECURE_STORE_SERVICE, httpService));
+
+    // Set up the mock objects
+    mockInMemoryOperationRunner = Mockito.mock(InMemorySourceControlOperationRunner.class);
+  }
+  
+  @Before
+  public void resetMock() throws Exception {
+    Mockito.reset(mockInMemoryOperationRunner);
+    PowerMockito.spy(TaskUtil.class);
```

Comment:
Static method mocking is a code smell. We should avoid it on new code

Reply:
removed, now the tests will run the actual tasks again a test bare repo